### DISCUSSION TITLE
Query returning all rows of related table ignoring WHERE condition

### DISCUSSION THREAD

**Discussion by usaadi** [supabase/supabase#22143] [Questions] [answered]
Hello,

I'm using Supabase cloud in a nextjs project. I have the following function that performs a select query using the Supabase client:

```
type Client = SupabaseClient<Database>;

/**
 * @name getRequestsByOrganizationUId
 * @description Get all the requests for the organization
 */
export function getRequestsByOrganizationUId(
  client: Client,
  organizationUid: string,
) {
  console.log('getRequestsByOrganizationUId: ', organizationUid);
  return client
    .from(REQUESTS_TABLE)
    .select<
      string,
      UrlRequest
    >(`id, uuid, request_url, organizations:organization_id (uuid)`)
    .eq('organizations.uuid', organizationUid)
    .order('created_at', { ascending: false })
    .throwOnError();
}
```
The requests table has a foreign key column "organization_id" which is related to the "id" column of the organizations table.

I want to filter the requests using a uuid column "uuid" of the organizations table.

Is the above function using the correct syntax to do this? If not, what is the correct way?

Thanks,
Osama

**Comment by GaryAustin1** [comment, marked as answer]
You need to use !inner to filter the main table rows on the fk being there. https://supabase.com/docs/reference/javascript/select?example=querying-referenced-table-with-inner-join

So organization_id!inner in the select.
